---
title: 'Cancer classification using plasma cirDNA: A small N and large p environment'
output: html_document
fontsize: 12pt
published: true
status: publish
mathjax: true
---

Over the last fifteen years the field of biology has undergone a significant cultural change. The pipette is being replaced by the piping operator. At the recent [Software Carpentry](https://software-carpentry.org/) workshop that occurred at Queen's University this week, I noticed that most of the people there to learn about UNIX programming were biologists[[^1]]. As the cost of high-throughput sequencing technology has fallen, the possibility of extracting important patterns, or biomarkers, from these high-dimensional data sets has increased. However, the need for robust statistical procedures for understanding and mining these data sets remains more important than ever. This post will use a machine learning approach for estimating classification accuracy in a small $N$ large $p$ environment, using data from an interesting 2012 paper, [Epigenetic markers of prostate cancer in plasma circulating DNA](https://www.ncbi.nlm.nih.gov/pubmed/22619380) by *Coretese t. al (2012)*. The goal of this post will be twofold: (1) outline the theoretical challenges in approximating out-of-sample accuracy in small $N$ large $p$ environments, and (2) construct a machine learning pipeline that determines the best model for cancer classification using a cirDNA data set with 4000+ features and only 39 observations.

## Introduction and data exploration

In *Cortese et. al (2012)* the researchers were interested in whether modifications to circulating DNA (cirDNA) contained biomarkers for prostate cancer patients. Were such patterns to exist, a non-invasive biomarker could potentially be used for assessing prostate cancer risk. The paper took a multi-faceted approach and had several findings:

1. DNA modification to regions adjacent to the gene encoding ring finger protein 219 distinguished prostate cancer from benign hyperplasias with good sensitivity (61%) and specificity (71%).
2. Repetitive sequences indicated a highly statistically significant loss of DNA at the pericentromeric region of chromosome 10 in prostate cancer patients.
3. A machine-learning technique developed using multi-locus biomarkers was able to correctly distinguished prostate cancer samples from unaffected controls with 72% accuracy.

The last finding of the paper is of interest to this post. However, we will begin with an exploration of the data. One of the nice things about the field of genomics is that data used in academic papers is usually posted on the NCBI's GEO, with this study [being no exception](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE36195). Here is a quick overview of the steps used to bring in and tidy the data.

1. The `.gpr` files were loaded using the `read.maimages`, `backgroundCorrect`, and `normalizeWithinArrays` functions from the `limma` package in `R`. The last function generates a [red/green intensity ratio](https://en.wikipedia.org/wiki/MA_plot) (log base 2). The two replicates were then averaged per probe.
2. The microarray data was generated from a "UHN Human CpG 12K Array", and annotation data was downloadable from the [NCBI site](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL2040). Annotation data was matched to `TxDb.Hsapiens.UCSC.hg19.knownGene` from the `Homo.sapiens` package by finding common overlap in DNA sequences. Where matching was not perfect, the closest gene location was used.
3. Duplicate and unknown genes were dropped. This left more than 4200 features.

Figures 1A and 1B below are a [Volcano plot](https://en.wikipedia.org/wiki/Volcano_plot_(statistics)) and a heatmap similar to one seen in the original paper. The former shows the genes which are differentially modified as measured by a simple t-test between cancer and control samples. As multiple hypothesis tests are being conducted, the p-values need to be adjusted to take into account the [false discovery rate](https://en.wikipedia.org/wiki/False_discovery_rate)[[^2]]. The heatmap in Figure 1B shows some of the differentially modified genes that tend to be, relatively, under expressed in modification for cancer patients.

<p align="center">
<img src="/figures/gg_cirDNA.png">
</p>
